addons/purchase_mrp/models/mrp_bom.py

```diff
@@ -11,12 +11,13 @@ class MrpBom(models.Model):
     @api.constrains('product_id', 'product_tmpl_id', 'bom_line_ids', 'byproduct_ids', 'operation_ids')
     def _check_bom_lines(self):
         res = super()._check_bom_lines()
+        precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
         for bom in self:
             if all(not bl.cost_share for bl in bom.bom_line_ids):
                 continue
             if any(bl.cost_share < 0 for bl in bom.bom_line_ids):
                 raise UserError(_("Components cost share have to be positive or equals to zero."))
-            if sum(bom.bom_line_ids.mapped('cost_share')) != 100:
+            if round(sum(bom.bom_line_ids.mapped('cost_share')), precision) != 100:
```

Comment:
I would be in favor of `float_round` here
Also, I don't get the reason why you're using the DP of Product's UoM here ? It's a cost share, in percent, and its precision is hardcoded:
https://github.com/odoo/odoo/blob/5d61d7023b3b0a5e6a47ad71bb12d548c797b5f9/addons/mrp/models/stock_move.py#L158-L160
So you could do the same here

Reply:
yes yes you are right! I just got used to using DP whenever I encounter precision/rounding issues. can you please check now?